selenium/che-selenium-core/bin/webdriver.sh

```diff
@@ -875,69 +878,79 @@ testProduct() {
-        echo -e "[TEST] ${YELLOW} RUN STABLE TESTS${NO_COLOUR}"
-        echo -e "[TEST] ${YELLOW}---------------------------------------------------${NO_COLOUR}"
-        echo "[TEST]"
+    defineTestsScope $@
+    applyCustomOptions $@
+
+    if [[ ${COMPARE_WITH_CI} == true ]]; then
+        fetchActualResults $@
+    else
+        prepareToFirstRun
+
+        if [[ ${TEST_INCLUSION} == ${TEST_INCLUSION_STABLE} ]]; then
+            echo "[TEST]"
+            echo -e "[TEST] ${YELLOW}---------------------------------------------------${NO_COLOUR}"
+            echo -e "[TEST] ${YELLOW} RUN STABLE TESTS${NO_COLOUR}"
+            echo -e "[TEST] ${YELLOW}---------------------------------------------------${NO_COLOUR}"
+            echo "[TEST]"
+        fi
+
+        testProduct $@
     fi
 
-    testProduct $@
-fi
+    analyseTestsResults $@
+    generateFailSafeReport
+    printProposals $@
+    storeTestReport
+    printElapsedTime
 
-analyseTestsResults $@
-generateFailSafeReport
-printProposals $@
-storeTestReport
-printElapsedTime
+    if [[ ${TESTS_SCOPE} =~ -DrunSuite ]] \
+        && [[ $(fetchFailedTestsNumber) == 0 ]] \
+        && [[ ${COMPARE_WITH_CI} == false ]] \
+        && [[ ${TEST_INCLUSION} == ${TEST_INCLUSION_STABLE} ]]; then
 
-if [[ ${TESTS_SCOPE} =~ -DrunSuite ]] \
-      && [[ $(fetchFailedTestsNumber) == 0 ]] \
-      && [[ ${COMPARE_WITH_CI} == false ]] \
-      && [[ ${TEST_INCLUSION} == ${TEST_INCLUSION_STABLE} ]]; then
+        if [[ $(suiteContainsUnstableTests) != 0 ]]; then
+            echo "[TEST]"
+            echo "[TEST] Test suite '${ORIGIN_TESTS_SCOPE:11}' doesn't have tests which are marked as unstable."
+            echo "[TEST] No more tests will be run."
+            echo "[TEST]"
+            exit
+        fi
+
+        TEST_INCLUSION=${TEST_INCLUSION_UNSTABLE}
+        START_TIME=$(date +%s)
 
-    if [[ $(suiteContainsUnstableTests) != 0 ]]; then
         echo "[TEST]"
-        echo "[TEST] Test suite '${ORIGIN_TESTS_SCOPE:11}' doesn't have tests which are marked as unstable."
-        echo "[TEST] No more tests will be run."
         echo "[TEST]"
-        exit
-    fi
+        echo -e "[TEST] ${YELLOW}---------------------------------------------------${NO_COLOUR}"
+        echo -e "[TEST] ${YELLOW} RUN UNSTABLE TESTS${NO_COLOUR}"
+        echo -e "[TEST] ${YELLOW}---------------------------------------------------${NO_COLOUR}"
+        echo "[TEST]"
 
-    TEST_INCLUSION=${TEST_INCLUSION_UNSTABLE}
-    START_TIME=$(date +%s)
+        testProduct $@
+        generateFailSafeReport
+        storeTestReport
+        printElapsedTime
+    fi
+}
 
-    echo "[TEST]"
-    echo "[TEST]"
-    echo -e "[TEST] ${YELLOW}---------------------------------------------------${NO_COLOUR}"
-    echo -e "[TEST] ${YELLOW} RUN UNSTABLE TESTS${NO_COLOUR}"
-    echo -e "[TEST] ${YELLOW}---------------------------------------------------${NO_COLOUR}"
-    echo "[TEST]"
+set -e
```

Comment:
Could you, please, explain the reasons to use "set -e"?
IMHO handling errors explicitly makes code much more reliable and clear in case like the follow:
```
checkBuild() {
    mvn package
    [[ $? != 0 ]] && { exit 1; }		
}
```

Reply:
`set -e` makes the script fail on any non-zero return code. Pros are:
- you don't have to explicitly check that commands and functions finish gracefully
- if something works not as expected you immediately get exit and error and can create an issue or fix a bug as fast as possible

Reply:
`mvn` returns non-zero exit code if tests failed. So, using `set -e` forces stopping script execution immediately and prevents report generation.

Reply:
I haven't noticed that. I'll remove this directive and add explicit check back. Thank you!

Reply:
done